src/internal/m365/collection/drive/delta_tree_test.go

```diff
@@ -40,17 +41,663 @@ func (suite *DeltaTreeUnitSuite) TestNewNodeyMcNodeFace() {
+	expect assertNode,
+) int {
+	// ensure the nodes match
+	require.NotNil(t, node, "node does not exist in tree")
+	require.Equal(
+		t,
+		expect.self,
+		node,
+		"non-matching node")
+
+	// ensure the node has the expected number of children
+	assert.Len(
+		t,
+		node.children,
+		len(expect.children),
+		"node has expected number of children")
+
+	var nodeCount int
+
+	for _, expectChild := range expect.children {
+		t.Run(expectChild.self.id, func(_t *testing.T) {
+			nodeChild := node.children[expectChild.self.id]
+			require.NotNilf(
+				_t,
+				nodeChild,
+				"child must exist with id %q",
+				expectChild.self.id)
+
+			// ensure each child points to the current node as its parent
+			assert.Equal(
+				_t,
+				node,
+				nodeChild.parent,
+				"should point to correct parent")
+
+			// recurse
+			nodeCount += compareNodes(_t, nodeChild, expectChild)
+		})
+	}
+
+	return nodeCount + 1
+}
+
+type tombs []assertNode
+
+func entomb(nodes ...assertNode) tombs {
+	if len(nodes) == 0 {
+		return tombs{}
+	}
+
+	return append(tombs{}, nodes...)
+}
+
+func (ts tombs) compare(
+	t *testing.T,
+	tombstones map[string]*nodeyMcNodeFace,
+) {
+	require.Len(t, tombstones, len(ts), "count of tombstoned nodes")
+
+	for _, entombed := range ts {
+		zombey := tombstones[entombed.self.id]
+		require.NotNil(t, zombey, "tombstone must exist")
+		assert.Nil(t, zombey.parent, "tombstoned node should not have a parent reference")
+
+		t.Run("assert_tombstones/"+zombey.id, func(_t *testing.T) {
+			compareNodes(_t, zombey, entombed)
+		})
+	}
+}
+
+// unlike the prior test, this focuses entirely on whether or not the
+// tree produced by folder additions is correct.
+func (suite *DeltaTreeUnitSuite) TestFolderyMcFolderFace_SetFolder_correctTree() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	tree := newFolderyMcFolderFace(nil)
+	rootID := odConsts.RootID
+
+	set := func(
+		parentID, id, name string,
+		isPackage bool,
+	) {
+		err := tree.SetFolder(ctx, parentID, id, name, isPackage)
+		require.NoError(t, err, clues.ToCore(err))
+	}
+
+	assert.Nil(t, tree.root)
+	assert.Empty(t, tree.folderIDToNode)
+
+	// add the root
+	set("", rootID, odConsts.RootPathDir, false)
+
+	assert.NotNil(t, tree.root)
+	assert.Equal(t, rootID, tree.root.id)
+	assert.Equal(t, rootID, tree.folderIDToNode[rootID].id)
+
+	an(tree.root).compare(t, tree, true)
+
+	// add a child at the root
+	set(tree.root.id, "lefty", "l", false)
+
+	lefty := tree.folderIDToNode["lefty"]
+	an(
+		tree.root,
+		an(lefty),
+	).compare(t, tree, true)
+
+	// add another child at the root
+	set(tree.root.id, "righty", "r", false)
+
+	righty := tree.folderIDToNode["righty"]
+	an(
+		tree.root,
+		an(lefty),
+		an(righty),
+	).compare(t, tree, true)
+
+	// add a child to lefty
+	set(lefty.id, "bloaty", "bl", false)
+
+	bloaty := tree.folderIDToNode["bloaty"]
+	an(
+		tree.root,
+		an(lefty, an(bloaty)),
+		an(righty),
+	).compare(t, tree, true)
+
+	// add a child to righty
```

Comment:
nit: comment consistency with the code
```suggestion
	// add another child to lefty
```